src/client/src/components/table/table.tsx

```diff
@@ -138,61 +140,76 @@ export const Table = <T extends GridValidRowModel>({
       }
     : {};
 
+  const dataGridSx =
+    rows.length === 0 && noRowsLabel
+      ? {
+          "--DataGrid-overlayHeight": "0px",
+          "--height": "0px",
+          ...quickFilterStyles,
+          ...disabledStyles,
+          ...sx,
+          ...loadingStyles,
+        }
+      : {
+          "--DataGrid-overlayHeight": "44px",
+          ...quickFilterStyles,
+          ...disabledStyles,
+          ...sx,
+          ...loadingStyles,
+        };
+
   return (
-    <DataGrid
-      sx={{
-        "--DataGrid-overlayHeight": "44px",
-        ...quickFilterStyles,
-        ...disabledStyles,
-        ...sx,
-        ...loadingStyles,
-      }}
-      data-cy={dataCy ?? "data-table"}
-      columnHeaderHeight={44}
-      sortingOrder={["asc", "desc"]}
-      loading={isLoading ?? !rows?.length}
-      rowCount={actualRowCount}
-      rows={rows}
-      columns={adjustedWidthColumns}
-      getRowHeight={() => (rowAutoHeight ? "auto" : 44)}
-      onRowClick={onRowClick}
-      pageSizeOptions={[maxRowsPerPage]}
-      slots={{ toolbar: GridToolbar }}
-      slotProps={{
-        pagination: {
-          ActionsComponent: TablePaginationActions,
-        },
-        toolbar: {
-          csvOptions: { disableToolbarButton: true },
-          printOptions: { disableToolbarButton: true },
-          showQuickFilter: showQuickFilter,
-        },
-      }}
-      localeText={muiLocales[i18n.language]}
-      disableColumnSelector
-      disableRowSelectionOnClick
-      hideFooter={actualRowCount < maxRowsPerPage}
-      hideFooterSelectedRowCount
-      disableColumnFilter
-      disableColumnMenu={true}
-      disableDensitySelector
-      filterModel={filterModel}
-      getRowClassName={getRowClassName ?? defaultRowClassName}
-      onFilterModelChange={onFilterModelChange}
-      sortModel={sortModel}
-      onSortModelChange={onSortModelChange}
-      paginationModel={paginationModel}
-      onPaginationModelChange={onPaginationModelChange}
-      onColumnResize={handleColumnResize}
-      apiRef={apiRef}
-      paginationMode={paginationMode}
-      sortingMode={sortingMode}
-      checkboxSelection={checkboxSelection}
-      disableColumnSorting={disableColumnSorting}
-      isRowSelectable={isRowSelectable}
-      rowSelectionModel={rowSelectionModel}
-      onRowSelectionModelChange={onRowSelectionModelChange}
-      getRowId={getRowId}
-    />
+    <Stack gap={1.5}>
```

Comment:
PP: Wenn ich das richtig verstanden habe ist immer entweder das _DataGrid_ oder der _noRows_ text sichtbar. Ich hätte das anstatt des _Stack_ eher in ein Fragment `<></>` gepackt und mit `? :` das eine oder andere angezeigt.